### PR TITLE
Miscellaneous low-risk code cleanup

### PR DESCRIPTION
**Pull request by basil**
See self-review for description of changes.

### CODE REVIEW

File: src/main/java/org/jenkinsci/plugins/gitclient/CliGitAPIImpl.java [hunk 1]
```diff
@@ -827,14 +827,12 @@ else if (!referencePath.isDirectory())
                             listener.getLogger().println("[WARNING] Reference path does not contain an objects directory (not a git repo?): " + objectsPath);
                         else {
                             File alternates = new File(workspace, ".git/objects/info/alternates");
-                            try (PrintWriter w = new PrintWriter(alternates, Charset.defaultCharset().toString())) {
+                            try (PrintWriter w = new PrintWriter(alternates, Charset.defaultCharset())) {
```
Comment (basil) [inline src/main/java/org/jenkinsci/plugins/gitclient/CliGitAPIImpl.java:830]: Using newer Java Platform functionality that avoids the messiness around `UnsupportedEncodingException`.

File: src/main/java/org/jenkinsci/plugins/gitclient/CliGitAPIImpl.java [hunk 2]
```diff
@@ -2321,7 +2319,7 @@ void fixSshKeyOnWindows(Path key) throws GitException {
      * NOT INTENDED FOR USE OUTSIDE THE PLUGIN.
      */
     @NonNull
-    private List<String> extraGitCommandArguments = Collections.<String>emptyList();
+    private List<String> extraGitCommandArguments = Collections.emptyList();
```
Comment (basil) [inline src/main/java/org/jenkinsci/plugins/gitclient/CliGitAPIImpl.java:2322]: Remove unnecessary type argument.

File: src/main/java/org/jenkinsci/plugins/gitclient/JGitAPIImpl.java [hunk 3]
```diff
@@ -1467,7 +1467,7 @@ else if (!referencePath.isDirectory())
                                     String absoluteReference = objectsPath.getAbsolutePath().replace('\\', '/');
                                     listener.getLogger().println("Using reference repository: " + reference);
                                     // git implementations on windows also use
-                                    try (PrintWriter w = new PrintWriter(alternates, "UTF-8")) {
+                                    try (PrintWriter w = new PrintWriter(alternates, StandardCharsets.UTF_8)) {
```
Comment (basil) [inline src/main/java/org/jenkinsci/plugins/gitclient/JGitAPIImpl.java:1470]: Using newer Java Platform functionality that avoids the messiness around `UnsupportedEncodingException`.

File: src/test/java/hudson/plugins/git/GitAPIBadInitTest.java [hunk 4]
```diff
@@ -51,7 +51,7 @@ public void testInitExistingDirectory() throws Exception {
     @Test
     public void testInitExistingFile() throws Exception {
         File existingFile = new File(tempDir, "file-exists");
-        FileUtils.writeStringToFile(existingFile, "git init should fail due to this file", "UTF-8");
+        Files.writeString(existingFile.toPath(), "git init should fail due to this file", StandardCharsets.UTF_8);
```
Comment (basil) [inline src/test/java/hudson/plugins/git/GitAPIBadInitTest.java:54]: Replacing Commons I/O with standard Java Platform functionality.

File: src/test/java/org/jenkinsci/plugins/gitclient/CliGitCommand.java [hunk 5]
```diff
@@ -58,9 +59,9 @@ private String[] run(boolean assertProcessStatus) throws IOException, Interrupte
         ByteArrayOutputStream bytesErr = new ByteArrayOutputStream();
         Launcher.ProcStarter p = launcher.launch().cmds(args).envs(env).stdout(bytesOut).stderr(bytesErr).pwd(dir);
         int status = p.start().joinWithTimeout(1, TimeUnit.MINUTES, listener);
-        String result = bytesOut.toString("UTF-8");
+        String result = bytesOut.toString(StandardCharsets.UTF_8);
         if (bytesErr.size() > 0) {
-            result = result + "\nstderr not empty:\n" + bytesErr.toString("UTF-8");
+            result = result + "\nstderr not empty:\n" + bytesErr.toString(StandardCharsets.UTF_8);
```
Comment (basil) [inline src/test/java/org/jenkinsci/plugins/gitclient/CliGitCommand.java:64]: Using newer Java Platform functionality that avoids the messiness around `UnsupportedEncodingException`.

File: src/test/java/org/jenkinsci/plugins/gitclient/CredentialsTest.java [hunk 6]
```diff
@@ -188,7 +188,7 @@ public void disableSETSID() {
     private BasicSSHUserPrivateKey newPrivateKeyCredential(String username, File privateKey) throws IOException {
         CredentialsScope scope = CredentialsScope.GLOBAL;
         String id = "private-key-" + privateKey.getPath() + random.nextInt();
-        String privateKeyData = FileUtils.readFileToString(privateKey, StandardCharsets.UTF_8);
+        String privateKeyData = Files.readString(privateKey.toPath(), StandardCharsets.UTF_8);
```
Comment (basil) [inline src/test/java/org/jenkinsci/plugins/gitclient/CredentialsTest.java:191]: Replacing Commons I/O with standard Java Platform functionality.

File: src/test/java/org/jenkinsci/plugins/gitclient/FilePermissionsTest.java [hunk 7]
```diff
@@ -108,7 +109,7 @@ private static void verifyFile(File repo, int staticPerm) throws IOException {
         String fileName = String.format("git-%03o.txt", staticPerm);
         File file = new File(repo, fileName);
         assertTrue("Missing " + file.getAbsolutePath(), file.exists());
-        String content = FileUtils.readFileToString(file, "UTF-8");
+        String content = Files.readString(file.toPath(), StandardCharsets.UTF_8);
```
Comment (basil) [inline src/test/java/org/jenkinsci/plugins/gitclient/FilePermissionsTest.java:112]: Replacing Commons I/O with standard Java Platform functionality.

File: src/test/java/org/jenkinsci/plugins/gitclient/FilePermissionsTest.java [hunk 8]
```diff
@@ -157,7 +158,7 @@ private void addFile() throws IOException, GitException, InterruptedException {
         String content = fileName + " and UUID " + UUID.randomUUID();
         File added = new File(repo, fileName);
         assertFalse(fileName + " already exists", added.exists());
-        FileUtils.writeStringToFile(added, content, "UTF-8");
+        Files.writeString(added.toPath(), content, StandardCharsets.UTF_8);
```
Comment (basil) [inline src/test/java/org/jenkinsci/plugins/gitclient/FilePermissionsTest.java:161]: Replacing Commons I/O with standard Java Platform functionality.

File: src/test/java/org/jenkinsci/plugins/gitclient/FilePermissionsTest.java [hunk 9]
```diff
@@ -174,7 +175,7 @@ private void modifyFile() throws IOException, GitException, InterruptedException
         String content = fileName + " chg UUID " + UUID.randomUUID();
         File modified = new File(repo, fileName);
         assertTrue(fileName + " doesn't exist", modified.exists());
-        FileUtils.writeStringToFile(modified, content, "UTF-8");
+        Files.writeString(modified.toPath(), content, StandardCharsets.UTF_8);
```
Comment (basil) [inline src/test/java/org/jenkinsci/plugins/gitclient/FilePermissionsTest.java:178]: Replacing Commons I/O with standard Java Platform functionality.

File: src/test/java/org/jenkinsci/plugins/gitclient/GitAPICliGitNotIntializedTest.java [hunk 10]
```diff
@@ -180,14 +180,14 @@ private void assertSubmoduleContents(File repo) throws IOException {
 
     private void assertFileContains(File file, String expectedContent) throws IOException {
         assertFileExists(file);
-        final String fileContent = FileUtils.readFileToString(file, "UTF-8");
+        final String fileContent = Files.readString(file.toPath(), StandardCharsets.UTF_8);
```
Comment (basil) [inline src/test/java/org/jenkinsci/plugins/gitclient/GitAPICliGitNotIntializedTest.java:183]: Replacing Commons I/O with standard Java Platform functionality.

File: src/test/java/org/jenkinsci/plugins/gitclient/GitAPICliGitNotIntializedTest.java [hunk 11]
```diff
@@ -180,14 +180,14 @@ private void assertSubmoduleContents(File repo) throws IOException {
...
         final String message = file + " does not contain '" + expectedContent + "', contains '" + fileContent + "'";
         assertTrue(message, fileContent.contains(expectedContent));
     }
 
     private void assertFileContents(File file, String expectedContent) throws IOException {
         assertFileExists(file);
-        final String fileContent = FileUtils.readFileToString(file, "UTF-8");
+        final String fileContent = Files.readString(file.toPath(), StandardCharsets.UTF_8);
```
Comment (basil) [inline src/test/java/org/jenkinsci/plugins/gitclient/GitAPICliGitNotIntializedTest.java:190]: Replacing Commons I/O with standard Java Platform functionality.

File: src/test/java/org/jenkinsci/plugins/gitclient/GitAPINotIntializedTest.java [hunk 12]
```diff
@@ -241,14 +241,14 @@ private void assertSubmoduleContents(File repo) throws IOException {
 
     private void assertFileContains(File file, String expectedContent) throws IOException {
         assertFileExists(file);
-        final String fileContent = FileUtils.readFileToString(file, "UTF-8");
+        final String fileContent = Files.readString(file.toPath(), StandardCharsets.UTF_8);
```
Comment (basil) [inline src/test/java/org/jenkinsci/plugins/gitclient/GitAPINotIntializedTest.java:244]: Replacing Commons I/O with standard Java Platform functionality.

File: src/test/java/org/jenkinsci/plugins/gitclient/GitAPINotIntializedTest.java [hunk 13]
```diff
@@ -241,14 +241,14 @@ private void assertSubmoduleContents(File repo) throws IOException {
...
         final String message = file + " does not contain '" + expectedContent + "', contains '" + fileContent + "'";
         assertTrue(message, fileContent.contains(expectedContent));
     }
 
     private void assertFileContents(File file, String expectedContent) throws IOException {
         assertFileExists(file);
-        final String fileContent = FileUtils.readFileToString(file, "UTF-8");
+        final String fileContent = Files.readString(file.toPath(), StandardCharsets.UTF_8);
```
Comment (basil) [inline src/test/java/org/jenkinsci/plugins/gitclient/GitAPINotIntializedTest.java:251]: Replacing Commons I/O with standard Java Platform functionality.

File: src/test/java/org/jenkinsci/plugins/gitclient/GitAPITest.java [hunk 14]
```diff
@@ -964,7 +964,7 @@ public void testMergeStrategy() throws Exception {
         testGitClient.add("file");
         testGitClient.commit("commit2");
         testGitClient.merge().setStrategy(MergeCommand.Strategy.OURS).setRevisionToMerge(testGitClient.getHeadRev(testGitDir.getAbsolutePath(), "branch1")).execute();
-        assertEquals("merge didn't selected OURS content", "content2", FileUtils.readFileToString(f, "UTF-8"));
+        assertEquals("merge didn't selected OURS content", "content2", Files.readString(f.toPath(), StandardCharsets.UTF_8));
```
Comment (basil) [inline src/test/java/org/jenkinsci/plugins/gitclient/GitAPITest.java:967]: Replacing Commons I/O with standard Java Platform functionality.

File: src/test/java/org/jenkinsci/plugins/gitclient/GitAPITestUpdate.java [hunk 15]
```diff
@@ -245,7 +244,7 @@ void touch(String path) throws IOException {
          */
         File touch(String path, String content) throws IOException {
             File f = file(path);
-            FileUtils.writeStringToFile(f, content, "UTF-8");
+            Files.writeString(f.toPath(), content, StandardCharsets.UTF_8);
```
Comment (basil) [inline src/test/java/org/jenkinsci/plugins/gitclient/GitAPITestUpdate.java:247]: Replacing Commons I/O with standard Java Platform functionality.

File: src/test/java/org/jenkinsci/plugins/gitclient/GitClientMaintenanceTest.java [hunk 16]
```diff
@@ -255,7 +254,7 @@ public void test_loose_objects_maintenance() throws Exception {
         // Confirm loose-object pack file is present in the pack directory
         File looseObjectPackFilePath = new File(objectsPath.getAbsolutePath(), "pack");
         String[] looseObjectPackFile = looseObjectPackFilePath.list((dir1, name) -> name.startsWith("loose-"));
-        collector.checkThat("Missing expected loose objects in git dir, only found " + Arrays.stream(looseObjectPackFile).collect(Collectors.joining(",")),
+        collector.checkThat("Missing expected loose objects in git dir, only found " + String.join(",", looseObjectPackFile),
```
Comment (basil) [inline src/test/java/org/jenkinsci/plugins/gitclient/GitClientMaintenanceTest.java:257]: More concise version of the same logic.

File: src/test/java/org/jenkinsci/plugins/gitclient/LegacyCompatibleGitAPIImplTest.java [hunk 17]
```diff
@@ -93,7 +93,7 @@ private void assertIsGitRepo(File dir) {
 
     private File touch(String path, String content) throws IOException {
         File f = new File(repo, path);
-        FileUtils.writeStringToFile(f, content, "UTF-8");
+        Files.writeString(f.toPath(), content, StandardCharsets.UTF_8);
```
Comment (basil) [inline src/test/java/org/jenkinsci/plugins/gitclient/LegacyCompatibleGitAPIImplTest.java:96]: Replacing Commons I/O with standard Java Platform functionality.

File: src/test/java/org/jenkinsci/plugins/gitclient/NetrcTest.java [hunk 18]
```diff
@@ -82,17 +81,13 @@ private void assertCredentials(TestHost host, Credentials cred)
 
     private void copyFileContents(String source, String destination) throws IOException
     {
-        try (InputStream sourceStream = Files.newInputStream(Paths.get(source));
-                OutputStream out = Files.newOutputStream(Paths.get(destination))) {
-            IOUtils.copy(sourceStream, out);
-        }
+        Files.copy(Paths.get(source), Paths.get(destination), StandardCopyOption.REPLACE_EXISTING);
```
Comment (basil) [inline src/test/java/org/jenkinsci/plugins/gitclient/NetrcTest.java:84]: Replacing Commons I/O with standard Java Platform functionality.

File: src/test/java/org/jenkinsci/plugins/gitclient/NetrcTest.java [hunk 19]
```diff
@@ -82,17 +81,13 @@ private void assertCredentials(TestHost host, Credentials cred)
...
     }
 
     private void copyResourceContents(String resource, String destination) throws IOException
     {
-        try (InputStream sourceStream = this.getClass().getClassLoader().getResourceAsStream(resource);
-                OutputStream out = Files.newOutputStream(Paths.get(destination))) {
-            IOUtils.copy(sourceStream, out);
+        try (InputStream sourceStream = this.getClass().getClassLoader().getResourceAsStream(resource)) {
+            Files.copy(sourceStream, Paths.get(destination), StandardCopyOption.REPLACE_EXISTING);
```
Comment (basil) [inline src/test/java/org/jenkinsci/plugins/gitclient/NetrcTest.java:90]: Replacing Commons I/O with standard Java Platform functionality.

**Review by MarkEWaite — APPROVED** [review]
Thanks!

**Review by MarkEWaite — APPROVED** [review]
Thanks!

**Pull request merged**